schema/sample/spectra/xray.md

```diff
@@ -1,7 +1,24 @@
 - xray (array<object>):
+  - sourceType (experiment|simulation|literature)
+  - sourceDetails (object):
+    - name (str): e.g., aiidalab.materialscloud.org
+    - uuid (str)
+    - doi (str)
   - cif (object):
     - filename (string)
   - pdb (object):
     - filename (string)
-  - rscrCode (string): for example, dia
+  - rscrCode (object): for example, dia
   - temperature (number, K)
+  - derivedProperties (object)
+    - poreProperties:
+      - sourceType (str)
```

Comment:
Why there is no sourceDetails here ?

I would add a object source instead of sourceType and sourceDetails.

```
- source (object)
  - type
  - name
  - doi
  - url
  - uuid ??? UUID to which database